code/client/src/Client.cpp

```diff
@@ -0,0 +1,87 @@
+#include <Client.h>
+
+bool Client::updateModule(){
+//    sendRequest();
+    return true;
+}
+
+void Client::sendRequest() {
+
+    yarp::os::Bottle outputBottle;
+    yarp::os::Bottle inputBottle;
+    outputBottle.addVocab(COLLATZ_VOCAB_REQ_ITEM);
+    outputBottle.addInt(n);
+    if (testOutcome == 0) {
+        std::cout << "Client side: Test unsuccessful. Sending 0 request" << std::endl;
+    }
+    std::cout << "Sending request " << n << std::endl;
+    clientPort.write(outputBottle,inputBottle);
+
+    if (inputBottle.get(0).asVocab() != COLLATZ_VOCAB_ITEM) {
+        std::cerr << "Client side: Incorrect Identifier received. Ignoring message," << std::endl;
+        return;
+    }
+    n = inputBottle.get(1).asInt();
+    t = inputBottle.get(2).asInt();
+    std::cout << "Received the pair (" << n << ", " << t << ")" << std::endl;
+    collatz_test(n,t);
+
+}
+
+void Client::collatz_test(int n, int t) {
+
+    if (t <= 0) {
+        std::cout << "Threshold 0 would not converge. Using 1 instead" << std::endl;
+        t = 1;
+    }
+    while (n > t){
+        if (n % 2 == 0){
+            n /= 2;
+        } else {
+            n *=3;
+            n++;
+        }
+    }
+    testOutcome = n;
```

Comment:
Why are you using the data member `testOutcome` to collect the result of the test instead of relying on the return value of this function?